### DISCUSSION TITLE
Hardware/SoC

### DISCUSSION THREAD

**Discussion by CiNcH83** [ludwig-v/wireless-carplay-dongle-reverse-engineering#55] [Q&A]
I don't really get the HW specs of those wireless CarPlay dongles. The log says that the SoC is a Freescale i.MX6(ARMv7). However when opening up those dongles (any version), SoC seems to be an Atmel AT91SAM9260. Can somebody clarify why the log states that it is an i.MX6? Wrong info? It is however also mentioned on the front page (README) of this repo.

**Comment by iDoka** [comment]
That's not wrong info.
HW indeed use i.MX6.
AT91SAM9260 has never supported DDr2/DDR3 SDRAM according to [official specs](https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-6221-32-bit-ARM926EJ-S-Embedded-Microprocessor-SAM9260_Datasheet.pdf).

I think by unknown reason SoC have camouflage/fake as AT91SAM9260.

**Reply by CiNcH83** [reply]
> AT91SAM9260 has never supported DDr2/DDR3 SDRAM

Ha! Nice catch! They really seem to be trying to hide something. So they might be pretty nervous about this repo which however helped out quite some people who bricked their dongles due to broken firmware with Carlinkit doing nothing to help out.

**Reply by crazyyi** [reply]
What's the cost of a Freescale i.MX6(ARMv7)? I checked online it seems to be around $40? Does it come with everything that powers the dongle, including RAM, WiFi and bluetooth chips?

**Reply by CiNcH83** [reply]
An i.MX6 UltraLite is around 4-10$ depending on the feature set and quantity. That's without flash, RAM and wireless chipset.

**Reply by crazyyi** [reply]
> An i.MX6 UltraLite is around 4-10$ depending on the feature set and quantity. That's without flash, RAM and wireless chipset.

Thanks. So with moderate RAM and flash, wireless chipsets is $30 maximum a good guess?

**Reply by CiNcH83** [reply]
Probably more towards 20$ for the overall hardware costs.

**Reply by crazyyi** [reply]
> Probably more towards 20$ for the overall hardware costs.

Wow can't believe it is that cheap. From your experience opening these dongles do you find any Apple certified chips in any of them? I heard that this kind of chips could cost much too, or at least just for meeting Apple's requirements.

**Reply by CiNcH83** [reply]
I doubt that this dongle is an Apple certified accessory. I fail to find any MFi circuitry on the board.
![image](https://user-images.githubusercontent.com/24539361/115681502-e1742e00-a354-11eb-8436-19558fa05887.png)
![image](https://user-images.githubusercontent.com/24539361/115681570-f2bd3a80-a354-11eb-8b66-d70c38b7796f.png)

**Reply by shadowswan** [reply]
The chip ending 3959 is likely a fake MFI authentication chip

https://upverter.com/part/upn/040421c077a11c52/

**Reply by CiNcH83** [reply]
Yes. Just found it too...
![image](https://user-images.githubusercontent.com/24539361/115682739-0cab4d00-a356-11eb-9a02-ccc2b21d1d58.png)

**Reply by shadowswan** [reply]
Yup that’s the one

**Reply by CiNcH83** [reply]
NXP/Freescale provide i.MX based MFi development kits. So I suppose this is legit. Surely this means additional royalties to Apple...

**Reply by crazyyi** [reply]
> The chip ending 3959 is likely a fake MFI authentication chip
> 
> https://upverter.com/part/upn/040421c077a11c52/

A fake one? Means it may or may not work in all situations? This is getting interesting. But a legit one would probably increase the cost a lot more. They need to have their own factory and 50+ workers to pass MFi's requirements.

**Reply by shadowswan** [reply]
I’m doubting apple is issuing them a license but may be they are sourcing these out the back door of another factory, looks like you can get the chips on AliExpress. Might not be a clone or fake chip and can’t see a proper data sheet so not sure how the license links in with it but appears it’s needed for the HU to identify it as a iPhone

**Comment by JJTech0130** [comment]
Ok, so I know this is a little old but someone I know has got their hands on a different model of CarlinKit, and it appears to not contain the 3959 chip? The PCB design appears to be slightly re-organized, but there is an empty footprint matching the pinout of the 3959 in roughly the same spot, and a new chip marked "3989A"? with more pins is occupying the previously unpopulated footprint in your picture, in mostly the same spot. Are they using multiple (fake?) MFi chip variants, or am I misunderstanding something?
![IMG_2357](https://github.com/ludwig-v/wireless-carplay-dongle-reverse-engineering/assets/53275876/d7ea9a38-6032-41d6-a1a9-67561880ed0c)
